nixos/modules/services/home-automation/homebridge.nix

```
@@ -0,0 +1,549 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.homebridge;
+
+  args = concatStringsSep " " (
+    [
+      "-U ${cfg.userStoragePath}"
+      "-P ${cfg.pluginPath}"
+      "--strict-plugin-resolution"
+    ]
+    ++ optionals cfg.allowInsecure [ "-I" ]
+  );
+
+  restartCommand = "sudo -n systemctl restart homebridge homebridge-config-ui-x";
+  homebridgePackagePath = "${pkgs.homebridge}/lib/node_modules/homebridge";
+
+  defaultConfigUIPlatform = {
+    inherit (cfg.uiSettings)
+      platform
+      name
+      port
+      standalone
+      restart
+      homebridgePackagePath
+      sudo
+      log
+      ;
+  };
+
+  defaultConfig = {
+    description = "Homebridge";
+    bridge = {
+      inherit (cfg.settings.bridge) name port;
+      # These have to be set at least once, otherwise the homebridge will not work
+      username = "CC:22:3D:E3:CE:30";
+      pin = "031-45-154";
+    };
+    platforms = [
+      defaultConfigUIPlatform
+    ];
+  };
+
+  defaultConfigFile = pkgs.writeTextFile {
+    name = "config.json";
+    text = builtins.toJSON defaultConfig;
+  };
```

I'm not sure if it's possible since you use the `defaultConfig`, but you might be able to do

```suggestion
  defaultConfigFile = settingsFormat.generate "config.json" defaultConfig;
```

Same goes for the `nixOverrideConfigFile`.

---

Made this change